[RM approach]: Adding dimensionability (#1760)
cntt-n/CNTT: doc/tech/approach.md

> This section introduces some networking related principles that should be followed during development and definition of Networking scopes for Reference Model, Reference Architecture 01 (OpenStack), Reference Architecture 02 (K8s) and down to Reference Implementations and Reference Conformance test suites under CNTT.

Reviewer: for consistency in use of uppercase and lowercase as well as some suggestion on language (without a "from" use of "down to"):
```suggestion
This section introduces some networking related principles that should be followed during development and definition of the networking scope for Reference Model, Reference Architectures, Reference Implementations and Reference Conformance test suites.
```

> -	**Abstraction**: A standardized layer between Virtualisation Layer domains and the Networking Hardware Layer resource pool hiding the implementation details of the Networking Hardware.

Reviewer: Reduce indentation to align with other bullets in the list.
```suggestion
- **Abstraction**: A standardized layer between Virtualisation Layer domains and the Networking Hardware Layer resource pool hiding the implementation details of the Networking Hardware.
```

> > **Note:** Abstraction may be deployed in many different ways e.g. depending on target use case requirements, workload characteristics, different algorithm implementions of pipeline stages and available platforms. This includes Network Functions running on server nodes with or without programmable hardware acceleration, or networking support functions running on programmable network switches in the network.

Reviewer: Not sure how abstraction can "be deployed". "Network Functions running on server nodes with or without programmable hardware acceleration ..." is not abstraction.

What is the purpose of this Note? Maybe then we can word it appropriately.

Author: @pgoyal01 I do not know. I did not write it. I agree that abstractions can not be deployed. What if I change it to implemented?

Reviewer: @CsatariGergely Can we first decide the purpose of this note -- what is it that we want to convey? Then we can word accordingly. Thanks

There are other terms such as "pipeline stages" (of what) that all need to be addressed.

Author: I'm happy to discuss this note, but this pr was supposed to be about something else. ...
For me th enote does not add any clarity. We could even remove it in a different pr.

- **Agnosticity** : Define Network Fabric concepts and models that can carry any type of traffic in terms of:
  - Control, User and Management traffic types

>   - Acceleration technologies supporting multiple types of Infrastructure deployments and Network Function workloads 

Reviewer: align tense with the "Agnosticity" tense:
```suggestion
  - Acceleration technologies that can support multiple types of Infrastructure deployments and Network Function workloads 
```

> -	**Automation**: Enable full automation, from Physical Fabric installation and provisioning to automation of VNF/CNF onboarding.

Reviewer: consistency
```suggestion
-	**Automation**: Enable full automation, from Physical Fabric installation and provisioning to automation of workloads (VNF/CNF) onboarding.
```

> -	**Openness**: All Networking is based on open source or standardized APIs (NBI and SBI) and should enable integration of Open Source networking components (e.g. SDN controllers).

Reviewer: nit
```suggestion
-	**Openness**: All Networking is based on open source or standardized APIs (NBI and SBI) and should enable integration of Open Source networking components such as SDN controllers.
```

> -	**Programmability**: Should enable a programmable forwarding plane controlled from a separately deployed control plane that abstracts into standardized functions and interfaces.

Reviewer: Under Agnosticity, control, user and management planes were specified and so for consistency:
```suggestion
-	**Programmability**: Should enable a programmableuser (forwarding) plane controlled from a separately deployed control plane utilising open-source/standardized functions and interfaces.
```

Author: I agree with the user plane suggestion, but I think the utilisation of open source and/or standard interfaces is not really clear.

Reviewer: @CsatariGergely the original wording included "standardized functions" -- I tried to generalise. I think the sentence is attempting to say that the "control" of the user plane utilises opensource and/or standadised functions and interfaces.

Author: Yes but this wording mixes the implementation using open source and the abstraction using standardized interfaces.

> -	**Scalability**: Should be scalable to handle all traffic traverse North-South and East-West enabling small up to large deployments.

Reviewer: ```suggestion
-	**Scalability**: Should be scalable to handle all traffic traverse North-South and East-West enabling small to large deployments.
```

- **Workload diversity**: Should be capable to provide connectivity to any type of workloads, including VNF , CNF and BareMetal workloads.

> -	**Telco Grade**: Should support deployments with Telco Grade characteristics requirements.

Reviewer: please use either characteristics or requirements

-	**Future proof**: Should support known and emerging technology trends e.g. SmartNICs, FPGAs and Programmable Switches, multi-cloud aspects (public/private/hybrid), multi-location aspects (multi-cluster, DCI, etc), Edge realted technologies, etc.

> - **Dimensionability**: Should support network design in a way that the network is capable to handle the traffic in a non-blocking manner without swapping out the leaf switches.

Reviewer: "leaf switches" implies a particular networking architecture. In the RM should we keep away from such decisions.
sentence simplification
```suggestion
- **Dimensionability**: Should support networks capable of handling the traffic in a non-blocking manner without swapping out networking hardware.
```

Reviewer 2: Didn't we plan to merge scalability and dimensionability? 
What about:
**Scalability** Should be scalable to handle all traffic traverse North-South and East-West enabling small to large deployments in a non-blocking manner.
